crates/cubecl-core/src/runtime_tests/minifloat.rs

```diff
@@ -0,0 +1,212 @@
+}
+
+pub fn test_scale<R: Runtime>(client: ComputeClient<R::Server, R::Channel>, vectorization: u8) {
+    if !ue8m0::is_supported(&client) {
+        println!("Unsupported, skipping");
+        return;
+    }
+
+    let data = [2.0, 1024.0, 57312.0, f32::from_bits(0x7F000000)];
+    let num_out = vectorization as usize;
+    let handle1 = client.create(f32::as_bytes(&data[..num_out]));
+    let handle2 = client.empty(num_out * size_of::<u8>());
+
+    unsafe {
+        kernel_scale::launch_unchecked::<R>(
+            &client,
+            CubeCount::Static(1, 1, 1),
+            CubeDim::new(1, 1, 1),
+            ArrayArg::from_raw_parts::<f32>(&handle1, num_out, vectorization),
+            ArrayArg::from_raw_parts::<u8>(&handle2, num_out, vectorization),
+        )
+    };
+
+    let actual = client.read_one(handle2.binding());
+    let actual = u8::from_bytes(&actual);
+    let expect: Vec<u8> = vec![0b1000_0000, 0b1000_1001, 0b1000_1111, 0b1111_1110];
+
+    // TODO: Eventually add approx comparison that can deal with arbitrary floats. Manually
+    // double check for now
+    let actual_2 = client.read_one(handle1.binding());
+    let actual_2 = f32::from_bytes(&actual_2);
+    println!("actual_2: {actual_2:?}");
+
+    assert_eq!(actual, &expect[..num_out]);
+    //assert_eq!(&actual_2[..num_out], &data[..num_out]);
+}
+
+#[allow(missing_docs)]
+#[macro_export]
+macro_rules! testgen_minifloat {
+    () => {
+        use super::*;
+
+        #[test]
+        fn test_fp8() {
+            let client = TestRuntime::client(&Default::default());
+            cubecl_core::runtime_tests::minifloat::test_fp8::<TestRuntime, FloatType>(
+                client.clone(),
+                1,
+            );
+            cubecl_core::runtime_tests::minifloat::test_fp8::<TestRuntime, FloatType>(
+                client.clone(),
+                2,
+            );
+            cubecl_core::runtime_tests::minifloat::test_fp8::<TestRuntime, FloatType>(
+                client.clone(),
+                4,
+            );
+        }
+
+        #[test]
+        fn test_fp6() {
+            let client = TestRuntime::client(&Default::default());
+            cubecl_core::runtime_tests::minifloat::test_fp6::<TestRuntime, FloatType>(
+                client.clone(),
+                1,
+            );
+            cubecl_core::runtime_tests::minifloat::test_fp6::<TestRuntime, FloatType>(
+                client.clone(),
+                2,
+            );
+            cubecl_core::runtime_tests::minifloat::test_fp6::<TestRuntime, FloatType>(
+                client.clone(),
+                4,
+            );
+        }
+
+        // #[test]
+        // fn test_fp4() {
+        //     let client = TestRuntime::client(&Default::default());
+        //     cubecl_core::runtime_tests::minifloat::test_fp4::<TestRuntime>(client.clone(), 2);
+        //     cubecl_core::runtime_tests::minifloat::test_fp4::<TestRuntime>(client.clone(), 4);
+        // }
```

Comment:
Dead code?

Reply:
Sort of, I'd leave it there as a reminder to add tests for fp4 once sub-byte types are properly supported